### DISCUSSION TITLE
Float2Str float scientific notation causes query error

### DISCUSSION THREAD

**Discussion by fizz-is-fish** [PyMySQL/mysqlclient#708] [Q&A] [answered]
When I use django orm to query, it will concatenate all float numbers with e0 by default. When the database field type is float, it will cause a query error.
For example, the following queries are not equivalent
select * from shipping_fee where height >= 2.0 limit 1;
select * from shipping_fee where height >= 2.0e0 limit 1;
Can it be optimized? If it is a normal float type, do not splice e0

**Comment by methane** [comment]
2.0 is not float, but decimal. 2.0e0 is float. So using 2.0 cause error.

**Reply by methane** [reply]
See #705.

**Reply by fizz-is-fish** [reply]
Could it be related to the mysql version? It is indeed not equal in 5.6. 
select * from a where weight >= 907.1800 order by weight asc;
select * from a where weight >= 907.18e0 order by weight asc;

The method of type conversion in very early versions of pymysql was as follows
```
def escape_float(value, mapping=None):
    return ('%.15g' % value)
```

**Reply by fizz-is-fish** [reply]
I tried versions 5.6 and 5.7 but neither worked. 8.0 and above are equivalent.

**Reply by fizz-is-fish** [reply]
Is it possible to add a piece of logic to the code to obtain the mysql version at startup. If it is 5.7 and previous versions, e0 will not be added. Otherwise, e0 will be added. As for obtaining the mysql version, you can initiate a sql query or obtain a parameter from the setting.py file. value

**Reply by methane** [reply, marked as answer]
No. floating point number always needs e0. There is no difference for MySQL 5.6.
It is your bug. Removing e0 from driver is bad fix.
Learn about decimal/floating number and float/double and fix your schema and SQL.

**Reply by fizz-is-fish** [reply]
ok, thx
